### PR TITLE
feat: add stealth appearance to button appearance enum

### PR DESCRIPTION
**Pull request by eljefe223**
# Description
Added `stealth` appearance to `<Button/>` `appearance` props.
Closes: #1598 

## Motivation & context
Stealth is a reoccurring pattern we see throughout Microsoft UI. This allows us to quickly repeat the pattern without having to copy styles over and over.

## Issue type checklist

- [ ] **Chore**: A change that does not impact distributed packages.
- [ ] **Bug fix**: A change that fixes an issue, link to the issue above.
- [x] **New feature**: A change that adds functionality.

**Is this a breaking change?**
- [ ] This change causes current functionality to break.

## Process & policy checklist

- [x] I have added tests for my changes.
- [x] I have tested my changes.
- [ ] I have updated the project documentation to reflect my changes.
- [x] I have read the [CONTRIBUTING](https://github.com/Microsoft/fast-dna/blob/master/CONTRIBUTING.md) documentation and followed the [standards](https://microsoft.github.io/fast-dna/docs/en/contributing/standards) for this project.

### CODE REVIEW

File: packages/fast-components-styles-msft/src/button/index.ts [hunk 1]
```diff
@@ -177,6 +177,21 @@ const styles: ComponentStyles<ButtonClassNameContract, DesignSystem> = (
             borderWidth: "0",
             justifyContent: "flex-start",
         },
+        button__stealth: {
+            minWidth: "32px",
+            maxWidth: "unset",
+            padding: "0",
```
Comment (bheston) [inline packages/fast-components-styles-msft/src/button/index.ts]: We shouldn't need to change anything other than colors here. They layout should be the same as the primary, secondary, outline, and lightweight styles.

Comment (eljefe223) [reply packages/fast-components-styles-msft/src/button/index.ts]: I think its open for discussion. The most common scenario for stealth button is it only displaying an Icon. The idea behind creating this appearance was to not have to override the width, padding and other properties every time we wanted this type of button. If someone wants a stealth button that displays like the other buttons they can just as easily add those overrides but that is not a common scenario.

File: packages/fast-components-class-name-contracts-msft/src/button/index.ts [hunk 2]
```diff
@@ -24,6 +24,11 @@ export interface ButtonClassNameContract extends BaseButtonClassNameContract {
      */
     button__justified?: string;
 
+    /**
+     * The stealth appearance modifier
```
Comment (nalogiudice) [inline packages/fast-components-class-name-contracts-msft/src/button/index.ts]: I wonder if we can add what "stealth" actually means here? Coming from the outside and not being a part of this design conversation, I'm not quite sure what stealth really means.

Comment (eljefe223) [reply packages/fast-components-class-name-contracts-msft/src/button/index.ts]: @nalogiudice I guess one could argue the same for the other appearances. This aligns with our color system. Maybe we need to add something in the Read Me that describes when to use each kind of appearance.

**Pull request merged**